src/react/branching.js

```diff
@@ -39,78 +39,74 @@ export type BranchStatusEnum = "ROOT" | "NO_BRANCH" | "NEW_BRANCH" | "BRANCH";
 // objects depending on various heuristics (if they have the same "type" for example)
 // A new branch state is created on a branch status of "NEW_BRANCH" and is reset to null once the branch is no
 // longer new
-export class BranchState {
-  constructor() {
-    this._branchesToValidate = [];
-  }
-  _branchesToValidate: Array<{
-    type: Value,
-    value: Value,
-  }>;
+export function getValueWithBranchingLogicApplied(realm: Realm, value: AbstractValue): Value {
```

Comment:
Can you explain this in more detail on an example? I never really understood what "applying branching logic" means in practice

Reply:
It means to add keys to ReactElement nodes in this specific branch.

Reply:
I also added a comment in my commit.